src/modules/site-v2/base/utils/tools.py

```diff
@@ -58,63 +63,77 @@ def lookup_report(kind, reportId, user=None, validate_user=True):
       raise ReportLookupError('You do not have access to that report.', 401)
 
   # If the user doesn't have permission to view this report, show an error message
-  if validate_user and not (report.username == user.name or user_is_admin()):
+  if validate_user and not (job.report.username == user.name or user_is_admin()):
     raise ReportLookupError('You do not have access to that report.', 401)
 
   # If all checks passed, return the report entity
-  return report
+  return job
 
 
-def try_submit(EntityClass, user, data, no_cache):
+def try_submit(kind, user, data, no_cache):
 
-  # Try submitting the job
   try:
```

Comment:
As part of the new pipeline architecture, `create`ing and `schedule`-ing a job are now separate processes.  Looking forward, this will allow us to define the Phenotype report as createable but not schedulable, since the results are computed in-place.